pkg/manifests/manifests.go

```diff
@@ -772,6 +774,94 @@ func (f *Factory) KubeStateMetricsDeployment() (*appsv1.Deployment, error) {
 	return d, nil
 }
 
+// KubeStateMetricsDenylistBoundsCheck ensures that the user is:
+// * able to enable metrics that are denied by default, and,
+// * unable to disable metrics that are enabled by default.
+func (f *Factory) KubeStateMetricsDenylistBoundsCheck(deployment *appsv1.Deployment, service *v1.Service, secret *v1.Secret) (*appsv1.Deployment, error) {
+	// Fail early if the deny-list is empty.
+	if len(f.config.ClusterMonitoringConfiguration.KubeStateMetricsConfig.MetricDenylist) == 0 {
+		return deployment, nil
+	}
+
+	// Fetch the `/metrics` port.
+	var p int32
+	for _, port := range service.Spec.Ports {
+		if port.Name == "https-main" {
+			p = port.Port
+			break
+		}
+	}
+
+	// Query the endpoint.
+	t := time.NewTimer(5 * time.Second)
```

Comment:
I'm usually not a huge fan of hardcoding times within the code, but I see the point. How about setting a var for this in any case? I know it's only used there but it'd be cool from a readability standpoint and for a latter change if needed.

Reply:
To add to this, is a Timer the right tool here? Wouldn't the worst case here be a hot loop creating http requests for 5 seconds? 
Wouldn't a Poll variant with timeout and maybe backoff make sense here?

Reply:
Done, PTAL.